Console.java

```diff
@@ -0,0 +1,189 @@
+
+    public Console(CommandLineOptions options, LineReader reader, Printer printer) {
+        this.options = options;
+        this.reader = reader;
+        this.printer = printer;
+    }
+
+    public void run() {
+        printer.info(COPYRIGHT);
+        try (Grakn.Client client = new GraknClient(options.address())) {
+            runRepl(client);
+        } catch (GraknClientException e) {
+            printer.error(e.getMessage());
+        }
+    }
+
+    private void runRepl(Grakn.Client client) {
+        while (true) {
+            ReplCommand command = ReplCommand.getCommand(reader, printer, "> ");
+            if (command instanceof ReplCommand.Exit) {
+                System.exit(0);
+            } else if (command instanceof ReplCommand.Help) {
+                printer.info(HELP_MENU);
+            } else if (command instanceof ReplCommand.Clear) {
+                reader.getTerminal().puts(InfoCmp.Capability.clear_screen);
+            } else if (command instanceof ReplCommand.DatabaseList) {
+                try {
+                    client.databases().all().forEach(database -> printer.info(database));
+                } catch (GraknClientException e) {
+                    printer.error(e.getMessage());
+                }
+            } else if (command instanceof ReplCommand.DatabaseCreate) {
+                try {
+                    client.databases().create(command.asDatabaseCreate().database());
+                    printer.info("Database '" + command.asDatabaseCreate().database() + "' created");
+                } catch (GraknClientException e) {
+                    printer.error(e.getMessage());
+                }
+            } else if (command instanceof ReplCommand.DatabaseDelete) {
+                try {
+                    client.databases().delete(command.asDatabaseDelete().database());
+                    printer.info("Database '" + command.asDatabaseDelete().database() + "' deleted");
+                } catch (GraknClientException e) {
+                    printer.error(e.getMessage());
+                }
+            } else if (command instanceof ReplCommand.Transaction) {
+                String database = command.asTransaction().database();
+                Grakn.Session.Type sessionType = command.asTransaction().sessionType();
+                Grakn.Transaction.Type transactionType = command.asTransaction().transactionType();
+                runTransactionRepl(client, database, sessionType, transactionType);
+            }
+        }
+    }
+
+    private void runTransactionRepl(Grakn.Client client, String database, Grakn.Session.Type sessionType, Grakn.Transaction.Type transactionType) {
+        try (Grakn.Session session = client.session(database, sessionType);
+             Grakn.Transaction tx = session.transaction(transactionType)) {
+            while (true) {
+                String prompt = database + "::" + sessionType.name().toLowerCase() + "::" + transactionType.name().toLowerCase() + "> ";
```

Comment:
does this have to be inside the loop? database, sessionType and transaction type are all set outside the loop right?

Reply:
Ok will move out.